l2-contracts/src/L2RewardManager.sol

```diff
@@ -23,22 +26,22 @@ import { L1RewardManagerStorage } from "mainnet-contracts/src/L1RewardManagerSto
 contract L2RewardManager is
     IL2RewardManager,
     L2RewardManagerStorage,
-    IXReceiver,
     AccessManagedUpgradeable,
-    UUPSUpgradeable
+    UUPSUpgradeable,
+    IOAppComposer
 {
     /**
-     * @notice xPufETH token on this chain
+     * @notice pufETH OFT token on this chain
      */
-    IERC20 public immutable XPUFETH;
+    IPufETH public immutable PUFETH;
 
     /**
-     * @notice Burner contract on Ethereum Mainnet
+     * @notice The rewards manager contract on L1
      */
     address public immutable L1_REWARD_MANAGER;
 
-    constructor(address xPufETH, address l1RewardManager) {
-        XPUFETH = IERC20(xPufETH);
+    constructor(address oft, address l1RewardManager) {
+        PUFETH = IPufETH(oft); // TODO: DO we really need this? We can use the oft directly
```

Comment:
it doesn't hurt to have it.
If you are somebody who is not that familiar with what OFT is, this gives you a clue :D